src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts

```diff
@@ -0,0 +1,206 @@
+
+  public formFieldProperties: any;
+
+  public categoryMasterList: any;
+
+  public framework: string;
+
+  public exists: boolean;
+
+  public formType = 'content';
+
+  public formAction = 'search';
+
+  public pageNumber: Number = 1;
+
+  public queryParams: any;
+
+  public redirectUrl: any;
+  /**
+ * formInputData is to take input data's from form
+ */
+  public formInputData: any;
+
+  refresh = true;
+  isShowFilterPlaceholder = true;
+  /**
+    * Constructor to create injected service(s) object
+    Default method of Draft Component class
+    * @param {Router} route Reference of Router
+    * @param {PaginationService} paginationService Reference of PaginationService
+    * @param {ConfigService} config Reference of ConfigService
+  */
+  constructor(config: ConfigService,
+    resourceService: ResourceService,
+    router: Router,
+    private activatedRoute: ActivatedRoute,
+    private _cacheService: CacheService,
+    private cdr: ChangeDetectorRef,
+    frameworkService: FrameworkService,
+    formService: FormService,
+    toasterService: ToasterService,
+
+  ) {
+    this.config = config;
+    this.resourceService = resourceService;
+    this.router = router;
+    this.frameworkService = frameworkService;
+    this.formService = formService;
+    this.toasterService = toasterService;
+    this.formInputData = {};
+  }
+
+  ngOnInit() {
+    this.frameworkService.initialize();
+    this.formInputData = {};
+    this.getQueryParams();
+    this.filterType = this.routerVal;
+    this.getMetaData();
+  }
+
+  getQueryParams() {
+    Observable
+      .combineLatest(
+      this.activatedRoute.params,
+      this.activatedRoute.queryParams,
+      (params: any, queryParams: any) => {
+        return {
+          params: params,
+          queryParams: queryParams
+        };
+      })
+      .subscribe(bothParams => {
+        if (bothParams.params.pageNumber) {
+          this.pageNumber = Number(bothParams.params.pageNumber);
+        }
+        this.queryParams = { ...bothParams.queryParams };
+        _.forIn(this.queryParams, (value, key) => {
+          if (typeof value === 'string' && key !== 'key') {
+            this.queryParams[key] = [value];
+          }
+        });
+        // console.log('this.queryParams', this.queryParams);
+        this.formInputData = _.pickBy(this.queryParams);
+        // console.log('this.queryParams', this.formInputData);
+      });
+  }
+  /**
+* getMetaData is gives form config data
+*/
+  getMetaData() {
+    this.frameworkService.frameworkData$.subscribe((frameworkData: Framework) => {
+      if (frameworkData && !frameworkData.err) {
+        this.categoryMasterList = _.cloneDeep(frameworkData.frameworkdata);
+        this.framework = frameworkData.framework;
+        this.exists = this._cacheService.exists(this.filterType + this.formAction);
+        if (this.exists) {
+          const data: any | null = this._cacheService.get(this.filterType + this.formAction);
```

Comment:
can we store processed data and retrieve the same. Instead of storing unprocessed data